code/controllers/Processes/diseases.dm

```diff
@@ -0,0 +1,44 @@
+var/datum/controller/process/diseases/disease_master
+
+/datum/controller/process/diseases
+	var/list/currentrun = list()
+	var/list/processing_diseases = list()
+
+/datum/controller/process/diseases/setup()
+	name = "disease"
+	schedule_interval = 20 // every 2 seconds
+	start_delay = 7
+	log_startup_progress("Disease controller starting.")
+	if(disease_master)
+		qdel(disease_master) //only one mob master
+	disease_master = src
+
+	register_diseases() //register all pre-round diseases created
+
+/datum/controller/process/diseases/statProcess()
+	..()
+	stat(null, "[processing_diseases.len] diseases")
+
+
+/datum/controller/process/diseases/doWork()
```

Comment:
Ahh, what I' saying is, that most disease processes use an istypeless `for` loops and not a `while` loop.